srcpkgs/nextcloud-client/template

```diff
@@ -0,0 +1,54 @@
+# Template file for 'nextcloud-client'
+pkgname=nextcloud-client
+version=2.6.0
+revision=1
+wrksrc="desktop-${version}"
+build_style=cmake
+configure_args="-Wno-dev"
+hostmakedepends="pkg-config"
+makedepends="qt5-tools-devel qt5-webengine-devel qt5-declarative-devel
+ qt5-webchannel-devel qt5-location-devel qtkeychain-qt5-devel sqlite-devel
+ qt5-webkit-devel libcloudproviders-devel $(vopt_if dolphin 'extra-cmake-modules kio-devel')"
+conf_files="/etc/Nextcloud/sync-exclude.lst"
+short_desc="NextCloud Desktop client"
+maintainer="yopito <pierre.bourgin@free.fr>"
+license="GPL-2.0-or-later"
+homepage="https://nextcloud.com/clients/"
+distfiles="https://github.com/nextcloud/desktop/archive/v${version}.tar.gz"
+checksum=7b3f3c14d2e44826a5183fd59a7412c6dd5ed00296873e35c566f75c14c3a3ea
+
+build_options="dolphin"
+desc_option_dolphin="Build KDE dolphin support"
+build_options_default="dolphin"
+
+case "$XBPS_TARGET_MACHINE" in
+	armv6*|armv7*) broken="qt5-tools-devel unavailable" ;;
```

Comment:
The package exists for arm too. What is the exact build error?

Reply:
I'm trying crossbuild for armv7l since you mentionned it, but `qt5` and even `sqlite` are unavailable for armv7l these days (probably related to #15745 ?):
```
=> nextcloud-client-2.6.0_1: the following build options are set:
   dolphin: Build KDE dolphin support (ON)
=> Installing armv7l cross pkg: cross-vpkg-dummy ...
=> nextcloud-client-2.6.0_1: building [cmake] ...
   [host] pkg-config-0.29.2_2: found (https://alpha.de.repo.voidlinux.org/current)
   [host] qt5-qmake-5.13.1_2: found (https://alpha.de.repo.voidlinux.org/current)
   [host] qt5-host-tools-5.13.1_2: found (https://alpha.de.repo.voidlinux.org/current)
   [host] qt5-tools-5.13.1_2: found (https://alpha.de.repo.voidlinux.org/current)
   [host] kcoreaddons-5.63.0_1: found (https://alpha.de.repo.voidlinux.org/current)
   [host] cmake-3.15.4_1: found (https://alpha.de.repo.voidlinux.org/current)
   [target] qt5-tools-devel-5.13.1_2: not found
   [target] qt5-webengine-devel-5.13.1_2: not found
   [target] qt5-declarative-devel-5.13.1_2: not found
   [target] qt5-webchannel-devel-5.13.1_2: not found
   [target] qt5-location-devel-5.13.1_2: not found
   [target] qtkeychain-qt5-devel-0.9.1_1: found (https://alpha.de.repo.voidlinux.org/current)
   [target] sqlite-devel-3.30.1_1: not found
   [target] qt5-webkit-devel-5.212.0_6: found (https://alpha.de.repo.voidlinux.org/current)
   [target] libcloudproviders-devel-0.3.0_3: found (https://alpha.de.repo.voidlinux.org/current)
   [target] extra-cmake-modules-5.63.0_1: found (https://alpha.de.repo.voidlinux.org/current)
   [target] kio-devel-5.63.0_1: found (https://alpha.de.repo.voidlinux.org/current)
=> qt5-5.13.1_2: the following build options are set:
   webengine: Build Chromium-based WebEngine component (OFF)
=> qt5-5.13.1_2: building [gnu-configure] (dependency of nextcloud-client) ...
```

Reply:
and while building `qt5`, libevent is unavailable too:
```
=> libevent-2.1.11_1: building [gnu-configure] (dependency of qt5) ...
   [target] libressl-devel-2.9.2_1: found (https://alpha.de.repo.voidlinux.org/current)
=> libevent-2.1.11_1: installing target dependencies: libressl-devel-2.9.2_1 ...
[...]
```

Reply:
exactly same rebuilds happens if crossbuilding on armv6l :(

Reply:
real root cause of massive rebuilds on arm* described in #15674

Reply:
I tested it again now that the repositories are unstaged. qt5-tools-devel is not an issue anymore however since we cannot build webegine on cross-compiled 32 bit platforms and the client requires it, it still has to be marked as broken. Please update explanation. The case statement could also be reduced to `arm*`.